src/discussions/posts/post-editor/PostEditor.jsx

```diff
@@ -0,0 +1,109 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { Button, Form } from '@edx/paragon';
+import { faComments, faQuestion, faStar } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { hidePostEditor } from '../data';
+
+import messages from './messages';
+
+function PostEditor({ intl }) {
+  const dispatch = useDispatch();
+  const cancelAdding = () => dispatch(hidePostEditor());
+
+  return (
+    <div className="mx-4 my-2">
+      <Form>
+        <Form.Group>
+          <Form.Label>
+            {intl.formatMessage(messages.type)}
+          </Form.Label>
+          <Form.Text className="mb-3" muted>
+            {intl.formatMessage(messages.type_description)} ({intl.formatMessage(messages.required)})
+          </Form.Text>
+
+          <Form.Check inline type="radio" id="question">
+            <Form.Check.Input type="radio" />
+            <Form.Check.Label>
+              <FontAwesomeIcon icon={faQuestion} />
+              <span className="ml-2">{intl.formatMessage(messages.question_type)}</span>
+            </Form.Check.Label>
+          </Form.Check>
+
+          <Form.Check inline type="radio" id="discussion">
+            <Form.Check.Input type="radio" />
+            <Form.Check.Label>
+              <FontAwesomeIcon icon={faComments} />
+              <span className="ml-2">{intl.formatMessage(messages.discussion_type)}</span>
+            </Form.Check.Label>
+          </Form.Check>
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>
+            {intl.formatMessage(messages.topic_area)}
+          </Form.Label>
+          <Form.Text id="topicAreaInput" muted>
+            {intl.formatMessage(messages.topic_area_description)} ({intl.formatMessage(messages.required)})
+          </Form.Text>
+          <Form.Control
+            as="select"
+            defaultValue="General"
+            aria-describedby="topicAreaInput"
+          >
+            <option>General</option>
```

Comment:
Could you add a TODO here to indicate that this needs to be filled in another PR.

Reply:
Done.